### PR TITLE
feat(onBlur): add blur event on the toggle

### PR DESCRIPTION
**Pull request by jare-talend**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**
Add the onBlur event on the Toggle component

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://361.talend.surge.sh)

File: packages/components/src/Toggle/Toggle.snapshot.test.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ jest.mock('react-dom');
 
 const defaultProps = {
 	id: 'id',
-	onChange: () => {},
+	onChange: jest.fn(),
```
Comment (jmfrancois) [inline packages/components/src/Toggle/Toggle.snapshot.test.js]: why do you need that if you don't test it ?

File: packages/forms/src/fields/BooleanField.js [hunk 2]
```diff
@@ -39,10 +40,16 @@ function BooleanField(props) {
 	const onChangeHandler = () => {
 		onChange(!formData, uiOptions);
 	};
+	const onBlurHandler = () => {
+		if (onBlur) {
+			onBlur(idSchema && idSchema.$id, formData);
```
Comment (jmfrancois) [inline packages/forms/src/fields/BooleanField.js]: why the content of the onBlur API is so different from the onChangeHandler ?

File: packages/forms/src/fields/BooleanField.js [hunk 3]
```diff
@@ -39,10 +40,16 @@ function BooleanField(props) {
 	const onChangeHandler = () => {
 		onChange(!formData, uiOptions);
 	};
+	const onBlurHandler = () => {
```
Comment (jmfrancois) [inline packages/forms/src/fields/BooleanField.js]: I'm pretty sure, you can add (event) => {

   onBlur(event, { id: odSchema.$id, formData }) would be a better API

For me event handler API is always (event, data)

File: packages/forms/src/fields/BooleanField.js (hunk 2 above)
Comment (jare-talend) [inline packages/forms/src/fields/BooleanField.js]: https://github.com/mozilla-services/react-jsonschema-form/blob/18ac59429e37859478e197039b8502fd07c8a556/src/components/widgets/BaseInput.js

indeed, It's weird but it is the defined format by react-jsonschema-form

File: packages/forms/src/fields/BooleanField.js (hunk 3 above)
Comment (jare-talend) [inline packages/forms/src/fields/BooleanField.js]: I can't if I want to match like  react-jsonschema-form

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://361.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://361.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://361.talend.surge.sh)

**Pull request merged**